### PR TITLE
feat(Neovel): add new presence

### PR DESCRIPTION
**Pull request by RisingSunLight42**
## Description 

Add a new presence for the novel website "Neovel".

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

neovel_proof_2
neovel_proof_1
neovel_proof_6
neovel_proof_5
neovel_proof_4
neovel_proof_3
gif_proof

</details>

### CODE REVIEW

File: websites/N/Neovel/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"settings": [
+		{
+			"id": "privacy",
+			"title": "Privacy Mode",
+			"icon": "fad fa-user-secret",
+			"value": false
+		},
+		{
+			"id": "time",
+			"title": "Show timestamp",
```
Comment (EncryptedDev) [inline websites/N/Neovel/dist/metadata.json:40]:
```suggestion
			"title": "Show Timestamp",
```

File: websites/N/Neovel/presence.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		delete presenceData.startTimestamp;
+		delete presenceDataSlide.startTimestamp;
+	}
+
+	switch (pathnameArray[1]) {
+		case "discover":
+			presenceData.details = "Discovering new books";
+			break;
+		case "tos":
+			presenceData.details = "Reading Neovel T.O.S.";
```
Comment (EncryptedDev) [inline websites/N/Neovel/presence.ts:31]:
```suggestion
			presenceData.details = "Reading Neovel ToS";
```

**Review by EncryptedDev — COMMENTED** [review]
Please attach a GIF of the slideshow working too

File: websites/N/Neovel/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "RisingSunLight",
+		"id": "240521747852558347"
+	},
+	"service": "Neovel",
+	"description": {
+		"en": "Designed by and for book lovers ! Discover countless new stories at your fingertips and join a strong positive community !",
+		"fr": "Pensé et conçu pour les amoureux des novels ! Accédez, lisez et découvrez simplement des milliers de novels et rejoignez une communauté de passionnés !",
```
Comment (Lavenderleaff) [inline websites/N/Neovel/dist/metadata.json:10]:
```suggestion
		"fr": "Conçu par et pour les amoureux des livres ! Découvrez d'innombrables nouvelles histoires à portée de main et rejoignez une forte communauté positive !",
```

**Review by lavylavenderr — COMMENTED** [review]
Provide an accurate French description.

File: websites/N/Neovel/presence.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+			if (pathnameArray.length > 2) presenceData.state = "Privacy Policy";
+			break;
+		case "book":
+			if (document.location.hostname.includes("neopload")) {
+				presenceData.details = "Viewing its book parameters";
```
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:36]: Well, in fact, idk how to say that in a concise way, that's why it's weird.
But to explain, it concerns the page where you add or remove tags, change descriptions, change cover, etc... Of the book.
Maybe I can change it to something like "Managing its book"?

**Review by itsmelouis — APPROVED** [review]

File: websites/N/Neovel/presence.ts [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+			if (!privacy) {
+				presenceData.state = {
+					0: "In its recents novels",
+					1: "In its lists",
+					2: "Reading its comments",
+				}[pathnameArray[2]];
+			}
+			break;
+		case "write":
+			presenceData.details = "Is going to write";
```
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:132]: Well, It's not evident, but on the site, the page with the "write" in the URL is a page before redirecting to the part of the site where you write, so you're not writing at this moment, but you're going to.

File: websites/N/Neovel/presence.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (EncryptedDev) [reply websites/N/Neovel/presence.ts:36]: yes

File: websites/N/Neovel/presence.ts [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+				if (!privacy) {
+					presenceData.state = `Book : ${
+						document.querySelector(`option[value="${pathnameArray[2]}"]`)
+							.textContent
+					} [${pathnameArray[3]}]`;
+					presenceData.largeImageKey = `https://neovel.io/V2/book/image?bookId=${pathnameArray[2]}&oldApp=false&imageExtension=1`;
+				}
+			} else {
+				presenceData.details = presenceDataSlide.details =
+					"Reading a book page";
```
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:46]: On this page, you don't read the book, but reading it's page. Maybe changing reading to viewing is more clear.
It's the page with the summary, the cover, the author, the description, etc...

File: websites/N/Neovel/presence.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (EncryptedDev) [reply websites/N/Neovel/presence.ts:132]: How long is the redirect page? No point supporting the page if it is just a short redirect

File: websites/N/Neovel/dist/metadata.json (hunk 3 above)
Comment (RisingSunLight42) [reply websites/N/Neovel/dist/metadata.json:10]: Well, the description of the site in the metadata is not good then xD

File: websites/N/Neovel/presence.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:46]: image
Looks like that

File: websites/N/Neovel/presence.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:132]: It depends on the user behavior, there's explanation on the system of paiment of the site and some comments from site's authors, and also key figures.

File: websites/N/Neovel/presence.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (RisingSunLight42) [reply websites/N/Neovel/presence.ts:46]: A book page example :
image

**Comment by RisingSunLight42** [comment]
> Please attach a GIF of the slideshow working too
I've also edited the presence description to add the gif.
gif_proof

File: websites/N/Neovel/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "RisingSunLight",
+		"id": "240521747852558347"
+	},
+	"service": "Neovel",
+	"description": {
+		"en": "Designed by and for book lovers ! Discover countless new stories at your fingertips and join a strong positive community !",
+		"fr": "Conçu par et pour les amoureux des livres ! Découvrez d'innombrables nouvelles histoires à portée de main et rejoignez une forte communauté positive !",
+		"es": "¡Hecho, diseñado y pensado para los amantes de las novelas! ¡Con la punta de tus dedos puedes acceder, leer & descubrir montones de historias nuevas y una comunidad fuerte!"
```
Comment (alexbcberio) [inline websites/N/Neovel/dist/metadata.json:11]: The ampersand (&) is very rarely, if its ever, used in Spanish. Please, change it by "y".

Comment (RisingSunLight42) [reply websites/N/Neovel/dist/metadata.json]: Resolved on 9729af17ed25f3d98a94f23e43dbc9ec23c31230

**Review by Bas950 — APPROVED** [review]
Nice, slideshows :) 👍

**Review by EncryptedDev — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Thank you for the PR 🎉

**Comment by RisingSunLight42** [comment]
> Thank you for the PR 🎉

No prob !

**Pull request merged**